src/decision/src/GpsDecision.cpp

```diff
@@ -17,19 +19,133 @@ GpsDecision::GpsDecision(int argc, char **argv, std::string node_name) {
 
     // Setup Subscriber(s)
     std::string gps_filtered_topic_name = "/gps_conversion/relative_gps";
+    std::string compass_topic_name = "/gps_conversion/compass";
     int refresh_rate = 10;
+    gps_subscriber = public_nh.subscribe(gps_filtered_topic_name, refresh_rate, &GpsDecision::gpsCurrentCallBack, this);
     gps_subscriber = public_nh.subscribe(gps_filtered_topic_name, refresh_rate, &GpsDecision::gpsCallBack, this);
-
+    compass_subcriber = public_nh.subscribe(compass_topic_name, refresh_rate, &GpsDecision::compassCallBack, this);
     // Setup Publisher(s)
     std::string twist_topic = public_nh.resolveName("command");
     uint32_t queue_size = 10;
     twist_publisher = nh.advertise<geometry_msgs::Twist>(twist_topic, queue_size);
 
 }
 
-// This is called whenever a new message is received
+double GpsDecision::distance(const geometry_msgs::Point::ConstPtr& relative_gps) {
+    //double current_x = relative_gps->x;
+    double current_x=0.0;
+    double current_y=0.0;
+    double current_z=0.0;
+
+    double x=relative_gps->x;
+    double y=relative_gps->y;
+    double z=relative_gps->z;
+
+    double distance=sqrt(pow((x-current_x),2)+pow((y-current_y),2)+pow((z-current_z),2));
+
+  return distance;
+}
+
+double GpsDecision::desiredAngle(const geometry_msgs::Point relative_gps,
+                                 float current_heading,geometry_msgs::Point currentPoint) {
+    const double TO_DEGREES=180/(M_PI);
+    const double TO_RADIANS=(M_PI)/180;
+
+    double next_x=relative_gps.x;
+    double next_y=relative_gps.y;
+    //calculate the relative x and y
+    double x=(next_x-currentPoint.x);
+    double y=(next_y-currentPoint.y);
+
+    //double dis=distance(relative_gps);
+    double tan=y/x;
```

Comment:
avoid naming variables like "tan", "sin", "cos", "abs", etc. as this may cause confusion and/or naming conflicts since there are mathematical functions with the same names